wgsl/index.bs

```diff
@@ -3581,6 +3600,15 @@ The `for(initializer; condition; continuing) { body }` statement is syntactic su
 * If `condition` is non-empty, it is checked at the beginning of the loop body and if unsatisfied then a [[#break-statement]] is executed.
 * If `continuing` is non-empty, it becomes a [[#continuing-statement]] at the end of the loop body.
 
+A declaration introduced in the body of the for loop is [=in scope=] until the
+end of the for loop's current iteration.
+That is, a variable declared in a loop statement is re-initialized each
+iteration that it is executed.
+A declaration introduced in the `initializer` of the for loop is [=in scope=]
```

Comment:
The for_header is really complicated grammar is really complicated.
I guess for later I'd like to see a refactor refactor to make named concepts:

     for_header
       :   for_init? SEMICOLON for_test? SEMICOLON for_update?

 and then definitions for for_init, for_test, for_update.

But that's probably too much for this PR.